### PR TITLE
{Scheduled-Query} Add scheduled query rules

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
Closed https://github.com/Azure/azure-cli/issues/17891

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Scheduled-Query

**Comment by yonzhan** [comment]
Scheduled-Query

File: src/scheduled-query/azext_scheduled_query/tests/latest/test_scheduled_query_scenario.py [hunk 1]
```diff
@@ -68,6 +73,12 @@ def test_scheduled_query(self, resource_group):
                      self.check('criteria.allOf[0].failingPeriods.minFailingPeriodsToAlert', 2),
                      self.check('criteria.allOf[0].failingPeriods.numberOfEvaluationPeriods', 3)
                  ])
+        self.cmd('monitor scheduled-query update -g {rg} -n {name1} --mad PT30M --auto-mitigate False --skip-query-validation True',
+                 checks=[
+                     self.check('skipQueryValidation', True),
+                     self.check('muteActionsDuration', '0:30:00'),
+                     self.check('autoMitigate', False)
+                 ])
```
Comment (kairu-ms) [inline src/scheduled-query/azext_scheduled_query/tests/latest/test_scheduled_query_scenario.py:81]: Could you add some tests for `actions`

File: src/scheduled-query/setup.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 # TODO: Confirm this is the right version number you want and it matches your
 # HISTORY.rst entry.
-VERSION = '0.3.1'
+VERSION = '0.3.2'
```
Comment (kairu-ms) [inline src/scheduled-query/setup.py:19]:
```suggestion
VERSION = '0.4.0'
```

File: src/scheduled-query/azext_scheduled_query/custom.py [hunk 3]
```diff
@@ -54,9 +79,23 @@ def list_scheduled_query(client, resource_group_name=None):
...
 
-def update_scheduled_query(cmd, instance, tags=None, disabled=False, condition=None, condition_query=None,
-                           description=None, actions=None, severity=None, window_size=None,
-                           evaluation_frequency=None, mute_actions_duration=None):
+def update_scheduled_query(cmd,
+                           instance,
+                           tags=None,
+                           disabled=None,
+                           condition=None,
+                           actions=None,
```
Comment (kairu-ms) [inline src/scheduled-query/azext_scheduled_query/custom.py:87]: The parameter "actions" is already deprecated. Please replace it by other arguments.

File: src/scheduled-query/azext_scheduled_query/_params.py [hunk 4]
```diff
@@ -35,4 +35,14 @@ def load_arguments(self, _):
         c.argument('mute_actions_duration', type=get_period_type(as_timedelta=True),
                    options_list=['--mute-actions-duration', '--mad'],
                    help='Mute actions for the chosen period of time (in ISO 8601 duration format) after the alert is fired.')
-        c.argument('actions', options_list=['--action', '-a'], action=ScheduleQueryAddAction, nargs='+', validator=get_action_group_validator('actions'))
+        c.argument('actions', options_list=['--action', '-a'], action=ScheduleQueryAddAction, nargs='+',
+                   validator=get_action_group_validator('actions'), deprecate_info=c.deprecate(redirect='--actions', hide=True))
```
Comment (kairu-ms) [inline src/scheduled-query/azext_scheduled_query/_params.py:39]: You should remove `ScheduleQueryAddAction` and `get_action_group_validator` as it is deprecated.

File: src/scheduled-query/azext_scheduled_query/_help.py [hunk 5]
```diff
@@ -16,12 +16,6 @@
 type: command
 short-summary: Create a scheduled query.
 parameters:
-  - name: --action -a
-    short-summary: Add an action group and optional webhook properties to fire when the alert is triggered.
-    long-summary: |
-        Usage:   --action ACTION_GROUP_NAME_OR_ID [KEY=VAL [KEY=VAL ...]]
-
-        Multiple action groups can be specified by using more than one `--action` argument.
```
Comment (kairu-ms) [inline src/scheduled-query/azext_scheduled_query/_help.py:24]: Please add help messages for action_groups and custom_properties.

Comment (PARADISSEEKR) [reply src/scheduled-query/azext_scheduled_query/_help.py:24]: Added help information in `_params.py`, do I still need to add it here?

Comment (kairu-ms) [reply src/scheduled-query/azext_scheduled_query/_help.py:24]: Yes. Please add help message for `custom_properties` with Usage

File: src/scheduled-query/azext_scheduled_query/custom.py [hunk 6]
```diff
@@ -25,27 +25,52 @@ def _build_criteria(condition, condition_query):
...
+                           condition_query=None,
+                           disabled=False,
+                           description=None,
+                           tags=None,
+                           location=None,
+                           severity=2,
+                           window_size='5m',
+                           evaluation_frequency='5m',
+                           target_resource_type=None,
+                           mute_actions_duration=None,
```
Comment (kairu-ms) [inline src/scheduled-query/azext_scheduled_query/custom.py:44]: The default value of `mute_actions_duration` is changed. Please add breaking change logs in history.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**